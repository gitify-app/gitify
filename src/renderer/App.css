html,
body,
#root {
  height: 100%;
  -webkit-user-select: none;
}

html::-webkit-scrollbar,
div.flex-grow::-webkit-scrollbar {
  width: 10px;
}

html::-webkit-scrollbar-thumb,
div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #c1c1c1;
  border-radius: 10px;
}

html::-webkit-scrollbar-thumb:hover,
div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #a8a8a8;
  border-radius: 10px;
}

html.dark::-webkit-scrollbar-track,
html.dark div.flex-grow::-webkit-scrollbar-track {
  background-color: #090e15;
}

html.dark::-webkit-scrollbar-thumb,
html.dark div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #24292e;
}

html.dark::-webkit-scrollbar-thumb:hover,
html.dark div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #3a3f44;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  margin: 0;
  cursor: default;
}

#nprogress {
  pointer-events: none;
}

#nprogress .bar {
  position: fixed;
  top: 0;
  left: 50px;
  z-index: 1031;
  background: #203354;

  width: 100%;
  height: 2px;
}

/**
 * Set emoji size according to surrounding text.
 * Ref: https://github.com/jdecked/twemoji?tab=readme-ov-file#inline-styles
 */
img.emoji {
  height: 1em;
  width: 1em;
  margin: 0 0.05em 0 0.1em;
  vertical-align: -0.1em;
}
