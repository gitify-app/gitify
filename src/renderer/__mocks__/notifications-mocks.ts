import type { AccountNotifications } from '../types';
import {
  mockEnterpriseNotifications,
  mockGitHubNotifications,
} from '../utils/api/__mocks__/response-mocks';
import {
  mockGitHubCloudAccount,
  mockGitHubEnterpriseServerAccount,
} from './state-mocks';

export const mockAccountNotifications: AccountNotifications[] = [
  {
    account: mockGitHubCloudAccount,
    notifications: mockGitHubNotifications,
    error: null,
  },
  {
    account: mockGitHubEnterpriseServerAccount,
    notifications: mockEnterpriseNotifications,
    error: null,
  },
];

export const mockSingleAccountNotifications: AccountNotifications[] = [
  {
    account: mockGitHubCloudAccount,
    notifications: [mockGitHubNotifications[0]],
    error: null,
  },
];
