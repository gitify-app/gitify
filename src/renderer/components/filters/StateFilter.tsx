import { type FC, useContext } from 'react';

import { BellIcon } from '@primer/octicons-react';
import { Stack, Text } from '@primer/react';

import { AppContext } from '../../context/App';
import type { FilterStateType } from '../../types';
import {
  FILTERS_STATE_TYPES,
  getStateDetails,
  getStateFilterCount,
  isStateFilterSet,
} from '../../utils/notifications/filters/state';
import { Checkbox } from '../fields/Checkbox';
import { Tooltip } from '../fields/Tooltip';
import { Title } from '../primitives/Title';

export const StateFilter: FC = () => {
  const { updateFilter, settings, notifications } = useContext(AppContext);

  return (
    <fieldset id="filter-state" className="mb-3">
      <Stack direction="horizontal" gap="condensed" align="baseline">
        <Title icon={BellIcon}>State</Title>
        <Tooltip
          name="tooltip-filter-state"
          tooltip={
            <Stack direction="vertical" gap="condensed">
              <Text>Filter notifications by state.</Text>
              <Text className="text-gitify-caution">
                ⚠️ This filter requires the{' '}
                <Text as="strong">Detailed Notifications</Text> setting to be
                enabled.
              </Text>
            </Stack>
          }
        />
      </Stack>

      <Stack direction="vertical" gap="condensed">
        {Object.keys(FILTERS_STATE_TYPES).map((stateType: FilterStateType) => {
          const stateDetails = getStateDetails(stateType);
          const stateTitle = stateDetails.title;
          const stateDescription = stateDetails.description;
          const isStateChecked = isStateFilterSet(settings, stateType);
          const stateCount = getStateFilterCount(notifications, stateType);

          return (
            <Checkbox
              key={stateType}
              name={stateTitle}
              label={stateTitle}
              checked={isStateChecked}
              onChange={(evt) =>
                updateFilter('filterStates', stateType, evt.target.checked)
              }
              tooltip={
                stateDescription ? <Text>{stateDescription}</Text> : null
              }
              disabled={!settings.detailedNotifications}
              counter={stateCount}
            />
          );
        })}
      </Stack>
    </fieldset>
  );
};
