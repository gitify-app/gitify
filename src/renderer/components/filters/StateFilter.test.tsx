import { act, fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { mockAccountNotifications } from '../../__mocks__/notifications-mocks';
import { mockSettings } from '../../__mocks__/state-mocks';
import { AppContext } from '../../context/App';
import type { SettingsState } from '../../types';
import { StateFilter } from './StateFilter';

describe('renderer/components/filters/StateFilter.tsx', () => {
  const updateFilter = jest.fn();

  describe('should render itself & its children', () => {
    it('with detailed notifications enabled', () => {
      const tree = render(
        <AppContext.Provider
          value={{
            settings: {
              ...mockSettings,
              detailedNotifications: true,
            } as SettingsState,
            notifications: mockAccountNotifications,
          }}
        >
          <MemoryRouter>
            <StateFilter />
          </MemoryRouter>
        </AppContext.Provider>,
      );

      expect(tree).toMatchSnapshot();
    });

    it('with detailed notifications disabled', () => {
      const tree = render(
        <AppContext.Provider
          value={{
            settings: {
              ...mockSettings,
              detailedNotifications: false,
            } as SettingsState,
            notifications: mockAccountNotifications,
          }}
        >
          <MemoryRouter>
            <StateFilter />
          </MemoryRouter>
        </AppContext.Provider>,
      );

      expect(tree).toMatchSnapshot();
    });
  });

  it('should be able to toggle user type - none already set', async () => {
    await act(async () => {
      render(
        <AppContext.Provider
          value={{
            settings: {
              ...mockSettings,
              filterStates: [],
            },
            notifications: [],
            updateFilter,
          }}
        >
          <MemoryRouter>
            <StateFilter />
          </MemoryRouter>
        </AppContext.Provider>,
      );
    });

    fireEvent.click(screen.getByLabelText('Open'));

    expect(updateFilter).toHaveBeenCalledWith('filterStates', 'open', true);

    expect(
      screen.getByLabelText('Open').parentNode.parentNode,
    ).toMatchSnapshot();
  });

  it('should be able to toggle user type - some filters already set', async () => {
    await act(async () => {
      render(
        <AppContext.Provider
          value={{
            settings: {
              ...mockSettings,
              filterStates: ['open'],
            },
            notifications: [],
            updateFilter,
          }}
        >
          <MemoryRouter>
            <StateFilter />
          </MemoryRouter>
        </AppContext.Provider>,
      );
    });

    fireEvent.click(screen.getByLabelText('Closed'));

    expect(updateFilter).toHaveBeenCalledWith('filterStates', 'closed', true);

    expect(
      screen.getByLabelText('Closed').parentNode.parentNode,
    ).toMatchSnapshot();
  });
});
