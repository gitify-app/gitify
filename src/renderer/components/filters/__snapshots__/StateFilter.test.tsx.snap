// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/components/filters/StateFilter.tsx should render itself & its children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <fieldset
        id="filter-state"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ"
          data-align="baseline"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <legend>
            <div
              class="Box-sc-g0xbh4-0 mb-2"
            >
              <div
                class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ"
                data-align="center"
                data-direction="horizontal"
                data-gap="condensed"
                data-justify="start"
                data-padding="none"
                data-wrap="nowrap"
                id="state"
              >
                <svg
                  aria-hidden="true"
                  class="octicon octicon-issue-opened"
                  display="inline-block"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  overflow="visible"
                  style="vertical-align: text-bottom;"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"
                  />
                  <path
                    d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z"
                  />
                </svg>
                <h2
                  class="Heading__StyledHeading-sc-1c1dgg0-0 eQYHUB"
                  sx="[object Object]"
                >
                  State
                </h2>
              </div>
            </div>
          </legend>
          <div
            aria-label="tooltip-filter-state"
            class="Box-sc-g0xbh4-0 relative"
            data-testid="tooltip-tooltip-filter-state"
            id="tooltip-filter-state"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-question text-gitify-tooltip-icon"
              display="inline-block"
              fill="currentColor"
              focusable="false"
              height="16"
              overflow="visible"
              style="vertical-align: text-bottom;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
              />
            </svg>
          </div>
        </div>
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ"
          data-align="stretch"
          data-direction="vertical"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
            data-align="center"
            data-direction="horizontal"
            data-gap="condensed"
            data-justify="start"
            data-padding="none"
            data-wrap="nowrap"
          >
            <input
              class="size-4 rounded-sm cursor-pointer"
              data-testid="checkbox-Draft"
              id="Draft"
              type="checkbox"
            />
            <label
              class="font-medium text-gitify-font cursor-pointer"
              for="Draft"
            >
              Draft
            </label>
            <span
              class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
            >
              0
            </span>
          </div>
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
            data-align="center"
            data-direction="horizontal"
            data-gap="condensed"
            data-justify="start"
            data-padding="none"
            data-wrap="nowrap"
          >
            <input
              class="size-4 rounded-sm cursor-pointer"
              data-testid="checkbox-Open"
              id="Open"
              type="checkbox"
            />
            <label
              class="font-medium text-gitify-font cursor-pointer"
              for="Open"
            >
              Open
            </label>
            <div
              aria-label="tooltip-Open"
              class="Box-sc-g0xbh4-0 relative"
              data-testid="tooltip-tooltip-Open"
              id="tooltip-Open"
            >
              <svg
                aria-hidden="true"
                class="octicon octicon-question text-gitify-tooltip-icon"
                display="inline-block"
                fill="currentColor"
                focusable="false"
                height="16"
                overflow="visible"
                style="vertical-align: text-bottom;"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
                />
              </svg>
            </div>
            <span
              class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
            >
              1
            </span>
          </div>
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
            data-align="center"
            data-direction="horizontal"
            data-gap="condensed"
            data-justify="start"
            data-padding="none"
            data-wrap="nowrap"
          >
            <input
              class="size-4 rounded-sm cursor-pointer"
              data-testid="checkbox-Merged"
              id="Merged"
              type="checkbox"
            />
            <label
              class="font-medium text-gitify-font cursor-pointer"
              for="Merged"
            >
              Merged
            </label>
            <span
              class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
            >
              0
            </span>
          </div>
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
            data-align="center"
            data-direction="horizontal"
            data-gap="condensed"
            data-justify="start"
            data-padding="none"
            data-wrap="nowrap"
          >
            <input
              class="size-4 rounded-sm cursor-pointer"
              data-testid="checkbox-Closed"
              id="Closed"
              type="checkbox"
            />
            <label
              class="font-medium text-gitify-font cursor-pointer"
              for="Closed"
            >
              Closed
            </label>
            <div
              aria-label="tooltip-Closed"
              class="Box-sc-g0xbh4-0 relative"
              data-testid="tooltip-tooltip-Closed"
              id="tooltip-Closed"
            >
              <svg
                aria-hidden="true"
                class="octicon octicon-question text-gitify-tooltip-icon"
                display="inline-block"
                fill="currentColor"
                focusable="false"
                height="16"
                overflow="visible"
                style="vertical-align: text-bottom;"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
                />
              </svg>
            </div>
            <span
              class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
            >
              0
            </span>
          </div>
          <div
            class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
            data-align="center"
            data-direction="horizontal"
            data-gap="condensed"
            data-justify="start"
            data-padding="none"
            data-wrap="nowrap"
          >
            <input
              class="size-4 rounded-sm cursor-pointer"
              data-testid="checkbox-Other"
              id="Other"
              type="checkbox"
            />
            <label
              class="font-medium text-gitify-font cursor-pointer"
              for="Other"
            >
              Other
            </label>
            <div
              aria-label="tooltip-Other"
              class="Box-sc-g0xbh4-0 relative"
              data-testid="tooltip-tooltip-Other"
              id="tooltip-Other"
            >
              <svg
                aria-hidden="true"
                class="octicon octicon-question text-gitify-tooltip-icon"
                display="inline-block"
                fill="currentColor"
                focusable="false"
                height="16"
                overflow="visible"
                style="vertical-align: text-bottom;"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
                />
              </svg>
            </div>
            <span
              class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
            >
              3
            </span>
          </div>
        </div>
      </fieldset>
    </div>
  </body>,
  "container": <div>
    <fieldset
      id="filter-state"
    >
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ"
        data-align="baseline"
        data-direction="horizontal"
        data-gap="condensed"
        data-justify="start"
        data-padding="none"
        data-wrap="nowrap"
      >
        <legend>
          <div
            class="Box-sc-g0xbh4-0 mb-2"
          >
            <div
              class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ"
              data-align="center"
              data-direction="horizontal"
              data-gap="condensed"
              data-justify="start"
              data-padding="none"
              data-wrap="nowrap"
              id="state"
            >
              <svg
                aria-hidden="true"
                class="octicon octicon-issue-opened"
                display="inline-block"
                fill="currentColor"
                focusable="false"
                height="16"
                overflow="visible"
                style="vertical-align: text-bottom;"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"
                />
                <path
                  d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z"
                />
              </svg>
              <h2
                class="Heading__StyledHeading-sc-1c1dgg0-0 eQYHUB"
                sx="[object Object]"
              >
                State
              </h2>
            </div>
          </div>
        </legend>
        <div
          aria-label="tooltip-filter-state"
          class="Box-sc-g0xbh4-0 relative"
          data-testid="tooltip-tooltip-filter-state"
          id="tooltip-filter-state"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-question text-gitify-tooltip-icon"
            display="inline-block"
            fill="currentColor"
            focusable="false"
            height="16"
            overflow="visible"
            style="vertical-align: text-bottom;"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ"
        data-align="stretch"
        data-direction="vertical"
        data-gap="condensed"
        data-justify="start"
        data-padding="none"
        data-wrap="nowrap"
      >
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
          data-align="center"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <input
            class="size-4 rounded-sm cursor-pointer"
            data-testid="checkbox-Draft"
            id="Draft"
            type="checkbox"
          />
          <label
            class="font-medium text-gitify-font cursor-pointer"
            for="Draft"
          >
            Draft
          </label>
          <span
            class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
          >
            0
          </span>
        </div>
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
          data-align="center"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <input
            class="size-4 rounded-sm cursor-pointer"
            data-testid="checkbox-Open"
            id="Open"
            type="checkbox"
          />
          <label
            class="font-medium text-gitify-font cursor-pointer"
            for="Open"
          >
            Open
          </label>
          <div
            aria-label="tooltip-Open"
            class="Box-sc-g0xbh4-0 relative"
            data-testid="tooltip-tooltip-Open"
            id="tooltip-Open"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-question text-gitify-tooltip-icon"
              display="inline-block"
              fill="currentColor"
              focusable="false"
              height="16"
              overflow="visible"
              style="vertical-align: text-bottom;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
              />
            </svg>
          </div>
          <span
            class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
          >
            1
          </span>
        </div>
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
          data-align="center"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <input
            class="size-4 rounded-sm cursor-pointer"
            data-testid="checkbox-Merged"
            id="Merged"
            type="checkbox"
          />
          <label
            class="font-medium text-gitify-font cursor-pointer"
            for="Merged"
          >
            Merged
          </label>
          <span
            class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
          >
            0
          </span>
        </div>
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
          data-align="center"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <input
            class="size-4 rounded-sm cursor-pointer"
            data-testid="checkbox-Closed"
            id="Closed"
            type="checkbox"
          />
          <label
            class="font-medium text-gitify-font cursor-pointer"
            for="Closed"
          >
            Closed
          </label>
          <div
            aria-label="tooltip-Closed"
            class="Box-sc-g0xbh4-0 relative"
            data-testid="tooltip-tooltip-Closed"
            id="tooltip-Closed"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-question text-gitify-tooltip-icon"
              display="inline-block"
              fill="currentColor"
              focusable="false"
              height="16"
              overflow="visible"
              style="vertical-align: text-bottom;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
              />
            </svg>
          </div>
          <span
            class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
          >
            0
          </span>
        </div>
        <div
          class="Stack__StyledStack-sc-x3xa2i-0 kooAIZ text-sm"
          data-align="center"
          data-direction="horizontal"
          data-gap="condensed"
          data-justify="start"
          data-padding="none"
          data-wrap="nowrap"
        >
          <input
            class="size-4 rounded-sm cursor-pointer"
            data-testid="checkbox-Other"
            id="Other"
            type="checkbox"
          />
          <label
            class="font-medium text-gitify-font cursor-pointer"
            for="Other"
          >
            Other
          </label>
          <div
            aria-label="tooltip-Other"
            class="Box-sc-g0xbh4-0 relative"
            data-testid="tooltip-tooltip-Other"
            id="tooltip-Other"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-question text-gitify-tooltip-icon"
              display="inline-block"
              fill="currentColor"
              focusable="false"
              height="16"
              overflow="visible"
              style="vertical-align: text-bottom;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.92 6.085h.001a.749.749 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.756 2.756 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6.24 6.24 0 0 0-.26.16.952.952 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661.103-.067.207-.132.313-.195l.007-.004c.1-.061.182-.11.258-.161a.969.969 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.612.612 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187a1.01 1.01 0 0 0-.34.398ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z"
              />
            </svg>
          </div>
          <span
            class="Text-sc-17v1xeu-0 kbcDeY px-1.5 py-0.75 rounded-full text-[10px] font-medium leading-none min-w-[16px] text-gitify-counter-text bg-gitify-counter-secondary"
          >
            3
          </span>
        </div>
      </div>
    </fieldset>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
